### ISSUE TITLE
The fate of Android SMAPI on 1.6

### ISSUE THREAD

**Issue by anonymus177** [ZaneYork/SMAPI-Android-Installer#372]
Hello @ZaneYork and thank you for your fantastic work. I'm sure this project has brought joy to thousands, if not even millions of people worldwide.

Now, my knowledge of coding is regrettably very basic, but from what I have understood about the upcoming update, it should minimize the code differences across various ports of Stardew Valley, meaning the Android version should be much more close to the original PC version (if I'm not mistaken!). 

Does this mean your work should be much less tedious or even not necessary at all? Or is it that significant changes will still be necessary due to how mods work, and not the game itself? And are you going to bother with finishing the SMAPI for 1.5 or rather wait for the new version if it dramatically simplifies the work that needs to be done?

Again, thank you for everything,
your dedicated fan.

**Comment by GonzaUwU** [comment]
It's not out yet for 1.5 and you're already thinking about 1.6 hahaha but I still support the work of @ZaneYork

**Comment by anonymus177** [comment]
Well, part of my question was dealing with the fact that possibly, we might not even see SMAPI finished on 1.5, since 1.6 should be released more or less at the same time on all platforms – and relatively soon.

**Comment by Pathoschild** [comment]
Hi! I'm the dev working on the modding changes in Stardew Valley 1.6.

The main difficulty for SMAPI-on-Android might be the migration from .NET 5 to .NET 6. That in itself should be pretty easy (unlike the difficult migration from .NET Framework 4 in Stardew Valley 1.5), but [.NET 6 finally supports Android natively](https://github.com/xamarin/xamarin-android/wiki/Migrating-Xamarin.Android-Applications-to-.NET-6) so the team which ports the game to Android might decide to drop Xamarin. That might require changes in SMAPI-on-Android, though in the long term it'd make updates much easier since PC and Android would be using the same framework.

Otherwise I'm not expecting any major changes to how the game works on Android (e.g. a lot of the differences are unavoidable to support features like tap-to-move), so we'll still need a SMAPI port on Android. 1.6 does have [many internal changes](https://stardewvalleywiki.com/Modding:Migrate_to_Stardew_Valley_1.6) to make things easier for mod authors, but most of those will either be handled in [the original SMAPI code](https://github.com/Pathoschild/SMAPI) or won't affect SMAPI (e.g. SMAPI itself doesn't care how item IDs work).

Keep in mind that 1.6 is still a long while away. It's in alpha on PC, which means it hasn't even started public beta and we haven't started porting it to other platforms like Android yet.

**tl;dr:** We'll still need a SMAPI port on Android, but updating to 1.6 should be much easier than updating to 1.5.

**Comment by anonymus177** [comment]
Hi @Pathoschild, thank you both for your answer and for your long-term care, dedication and work on the Stardew Valley modding scene. 

I imagine dropping Xamarin would be the obvious thing to do since native use of .NET 6 should be easier to utilize even by the porting team themselves. 

Anyway, this really cleared out all my speculations. I am glad Zane won't be burdened as much by the next update, regardless of how long it takes to complete.
